library/core/src/iter/traits/iterator.rs

```diff
@@ -307,10 +308,10 @@ pub trait Iterator {
     /// This method will eagerly skip `n` elements by calling [`next`] up to `n`
     /// times until [`None`] is encountered.
     ///
-    /// `advance_by(n)` will return [`Ok(())`][Ok] if the iterator successfully advances by
-    /// `n` elements, or [`Err(k)`][Err] if [`None`] is encountered, where `k` is the number
-    /// of elements the iterator is advanced by before running out of elements (i.e. the
-    /// length of the iterator). Note that `k` is always less than `n`.
+    /// `advance_by(n)` will return `Ok(())` if the iterator successfully advances by
+    /// `n` elements, or a `Err(NonZeroUsize)` with value `k` if [`None`] is encountered,
+    /// where `k` is remaining number of steps that could not be advanced because the iterator ran out.
+    /// Note that `k` is always less than `n`.
```

Comment:
I don't think this is correct?  Maybe something more like
```suggestion
    /// If `self` is empty and `n` is non-zero, then this returns `Err(n)`.  Otherwise, `k` is always less than `n`.
```

Reply:
fixed